integration/query_array_compat_test.go

```diff
@@ -20,42 +20,60 @@ import (
 
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/bson/primitive"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
 )
 
 func TestQueryArrayCompatSize(t *testing.T) {
+	setup.SkipForTigrisWithReason(t, "https://github.com/FerretDB/FerretDB/issues/908")
```

Comment:
Now is the question – why we skip that test if we fixed it, and arrays are just skipped for both target and compat systems when target uses Tigris?

Reply:
This isn't fixed. In shareddata we have `Composites` which contain mix types in the array. Because we [skip creating that collection](https://github.com/FerretDB/FerretDB/blob/main/integration/setup/setup.go#L151) for tigris in setup, when we query on compat tests we see mismatch. I'd say we leave tigris out for now.

Reply:
> Because we skip creating that collection for tigris in setup, when we query on compat tests we see mismatch

You are linking to the wrong function. `TestQueryArrayCompatSize` calls `testQueryCompat` that calls `SetupCompat` that calls `setupCompatCollections`, not `setupCollection`.

https://github.com/chilagrow/FerretDB/blob/3331480fa0c2ca01f7839caa9508d78fd769537a/integration/setup/setup_compat.go#L166-L172

That code target's handler for both target and compat systems, skipping creating collection in both system if it is not compatible with target system. There is no mismatch.

Reply:
This is interesting because, the test still continues to not work on tigris and my above assumption is wrong. I'll create an issue for investigating that.

Reply:
Great, let's change `SkipForTigrisWithReason`'s argument then; that test is skipped not because Tigris does not support arrays yet